### PR TITLE
Fix keymaps, RGB matrix, add support for WS2812

### PR DESCRIPTION
**Pull request by AnthonyNguyen168**
## Description

- Keymap is fixed, TO(4) is removed.
- Fix bug for RGB matrix, the Red and Green is reversed in hardware, this is a software improvisation.
- Add support for WS2812, these LEDs are controlled by Backlight keycode.

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

File: keyboards/canary/canary60rgb/keymaps/via/keymap.c [hunk 1]
```diff
@@ -21,20 +21,20 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
...
         _______,          _______, _______, _______, _______, _______, NK_TOGG, _______, _______, KC_END,  KC_PGDN, KC_VOLU, KC_MUTE,
-        _______, _______, TO(3),                            TO(4),                     _______, _______, KC_MPRV, KC_VOLD, KC_MNXT
+        _______, _______, TO(3),                              _______,                   _______, _______, KC_MPRV, KC_VOLD, KC_MNXT
     ),
     [2] = LAYOUT(
         _______, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_DEL,
         _______, RGB_TOG, RGB_MOD, RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, RGB_VAI, RGB_VAD, _______, _______, _______, _______, RESET,
         _______, _______, _______, _______, _______, _______, _______, _______, RGB_SPI, RGB_SPD, _______, _______,          EEP_RST,
-        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______,          BL_TOGG, BL_STEP, BL_INC,  BL_DEC,  BL_ON,   BL_OFF, _______, _______, _______, _______, _______, _______,
```
Comment (fauxpark) [inline keyboards/canary/canary60rgb/keymaps/via/keymap.c:37]: Please use the RGB keycodes, not Backlight.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/canary/canary60rgb/keymaps/via/keymap.c (hunk 1 above)
Comment (AnthonyNguyen168) [reply keyboards/canary/canary60rgb/keymaps/via/keymap.c:37]: Hi @fauxpark, I already used RGB keycode to control RGB matrix per key. We added RGB lighting WS2812 for this PCB and wanted to control RGB lighting and matrix separately. Therefore we wrote custom code to use Backlight keycodes to control RGB lighting.
If user want to used Backlight keycode, they would define their own function for these keycode
process_record_user(keycode, record))
I follow the guidance for keycode in this link:
https://beta.docs.qmk.fm/using-qmk/hardware-features/lighting/feature_rgblight

Please consider my changes.

Thanks.

Comment (fauxpark) [reply keyboards/canary/canary60rgb/keymaps/via/keymap.c:37]: That page says nothing about using Backlight keycodes to control RGB. IMO repurposing these keycodes only perpetuates the confusion between the different lighting features.

Eventaully the keycodes for RGBLight/RGB Matrix will be separated so they can be controlled independently, but for now you just have to create keyboard-level custom keycodes to control the other one.

Comment (AnthonyNguyen168) [reply keyboards/canary/canary60rgb/keymaps/via/keymap.c]: Hi @fauxpark,
Ok, I will not re-purpose Backlight keycode for RGBLight controlling.
Those code are removed.
Please consider my changes.
Thanks.

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Didn't catch it before, but this has both rgblight and rgb matrix enabled, and because of that, compiles too large

**Comment by AnthonyNguyen168** [comment]
> Didn't catch it before, but this has both rgblight and rgb matrix enabled, and because of that, compiles too large

I compiled on my local machine, the binary is ~96% of memory. I flashed the binary and it works normally.

**Pull request merged**